subsys/net/lib/download_client/src/http.c

```diff
@@ -130,31 +134,41 @@ static int http_header_parse(struct download_client *client, size_t *hdr_len)
 		client->buf[i] = tolower(client->buf[i]);
 	}
 
-	p = strstr(client->buf, "http/1.1 206");
-	if (!p) {
-		p = strstr(client->buf, "http/1.1 404");
-		if (p) {
-			LOG_ERR("Server response was 404: file not found");
-			return -1;
-		}
-		if (client->proto == IPPROTO_TLS_1_2
-		   || IS_ENABLED(CONFIG_DOWNLOAD_CLIENT_RANGE_REQUESTS)) {
-			LOG_ERR("Server did not honor partial content request");
-			return -1;
-		}
-		p = strstr(client->buf, "http/1.1 200");
-		if (!p) {
-			LOG_ERR("Server response is not 200 Success");
-			return -1;
+	/* Look for the status code just after "http/1.1 " */
+	p = strstr(client->buf, "http/1.1 ");
+	if (p == NULL) {
+		LOG_ERR("Server response missing HTTP/1.1");
+		return -1;
+	}
+	p += strlen("http/1.1 ");
+
+	const unsigned int http_status = strtoul(p, &q, 10);
```

Comment:
nit: I would drop `const` and move it at the beginning of the scope. I try to keep everything there, to avoid mixing declarations and code.

Reply:
I've been reading up on rust and it's affecting how I think.  I want everything to be immutable by default now.  :)